### PR TITLE
Renaming bandwidth variable in simulation

### PR DESCRIPTION
**Pull request by fschlueter**
Addresses #593

### CODE REVIEW

**Comment by cg-laser** [comment]
the calculation is correct. It is just the variable name that could be improved. See discussion in #593

**Comment by fschlueter** [comment]
@cg-laser, I am not sure that I fully understand this. The calculation of the v_rms is (from Christoph branch):
```
                    self.__v_rms_per_channel[station_id][channel_id] = (noise_temp_channel * 50 * scipy.constants.k *
                           self.__bandwidth_per_channel[station_id][channel_id] / units.Hz) ** 0.5  # from elog:1566 and https://en.wikipedia.org/wiki/Johnson%E2%80%93Nyquist_noise (last Eq. in "noise voltage and power" section
```
Looking at the wikipage, I think the appropriate formular is:
image
What happened to the factor "4"? I assume the factor "50" in the equation above refers to 50 Ohm? I still do not understand why integrating over the absolute amplifier response is the correct thing. What I can imagine is that one would need to amplify the noise power calculated as proposed in this PR with the amplifier gain and that this is equivalent with integrating over the amplifier response?

**Comment by fschlueter** [comment]
I think to remember that there was such a discussion one on the NuRadio slack. It's a shame that we do not have the history there.

**Comment by fschlueter** [comment]
this logging is certainly very confusing `logger.status(f"bandwidth of station {self._station_id} channel {channel_id} is {bandwidth/units.MHz:.1f}MHz") `

**Comment by fschlueter** [comment]
> With your approach, frequencies that are attenuated by the filter by a factor 100 contribute the same amount as frequencies that are not attenuated at all,
Hm, I agree that with "my approach" you would need to multiply again with the amplifier gain. But when you average this over the exact same bandwidth, does that really result in some other result?

**Comment by fschlueter** [comment]
@cg-laser @sjoerd-bouma. I reverted the calculation and just renamed the variable. However, now things are a bit inconsistent: We still have the variable `self._bandwidth` and we store everything still with this name. Also should the square  only enter the calculation in lines ~348?

**Review by cg-laser — COMMENTED** [review]
I don't think I prefer the new name. `filter_response` sounds more like the S21 parameter as a function of frequency. 
Wouldn't the better option be to keep the name "bandwidth" but add comments to the code that specifies "bandwidth refers to the detector response (filter, amplifiers, etc.) integrated over frequency"
and I would also improve the logger output to also mention that.

**Comment by fschlueter** [comment]
So it's the integrated filter/detector response. I can change `filter_response` to `integrated_filter_response` but `bandwidth` is more confusing I think. I can improve the logger output. What about the keywords in the hdf5 files?

**Comment by fschlueter** [comment]
> and I would also improve the logger output to also mention that.

Did you notice that I added a normalisation to the logger output such that it gives you a reasonable value for the bandwidth:
```
logger.debug(f"Estimated bandwidth of station {self._station_id} channel {channel_id} is "
                             f"{integrated_channel_response / mean_integrated_response / units.MHz:.1f} MHz")

```
With that I get something like this `STATUS:NuRadioMC:Estimated bandwidth of station 101 channel 0 is 784.6 MHz` (however I switched `status` to `debug`)

**Comment by fschlueter** [comment]
@cg-laser is that fine? Any other suggestions? I also updated the documentation.

**Review by cg-laser — COMMENTED** [review]
please don't switch to debug output. This is a useful output that let's the user check if the channels and noise temperatures and RMS values were calculated correctly. 

I don't like the normalization because it will give wrong answers for filters. 
Can you rename `mean_integrated_response` to `mean_amplification` 
And I would also set `mean_amplification` to 1 if the maximum amplification was 1. This allows to correctly detect if there is any amplification at all in the system. 

Why don't output 
"integrated channel response = X MHz, average amplification = X, maximum amplification = X, approximate bandwidth = X"

Or even better, if the `mean_amplification` is one, output what was outputted before. Only if we have amplification, output what I wrote above. 

For the documentation, I would rather not change the name of the field due to backward compatibility, but could you update the documentation saying that this is the integrated response which is equal to the bandwidth in case of no amplification

File: documentation/source/NuRadioMC/pages/HDF5_structure.rst [hunk 1]
```diff
@@ -72,14 +72,26 @@ The top-level attributes can be accessed using ``f.attrs``. These contain:
...
+The station-level attributes can be accessed using ``f[station_<station_id>].attrs``. The first two attributes ``Vrms`` and ``bandwidth`` also exist on the top-level and refer to the corresponding to the first station/channel pair.
+
+    .. _hdf5-station-attrs-table:
+
+    .. csv-table:: HDF5 station attributes
+            :header: "Key", "Description"
+            :widths: auto
+            :delim: |
+
+            ``Vrms`` | RMS of the voltage used as thermal noise floor :math:`v_{n} = (k_{B} \, R \, T \, \Delta f) ^ {0.5}`. See the relevant section "Noise voltage and power" in this `wiki article <https://en.wikipedia.org/wiki/Johnson%E2%80%93Nyquist_noise>`_ (last two equations). Determine from ``Tnoise`` and ``bandwidth`` (see below).
```
Comment (cg-laser) [inline documentation/source/NuRadioMC/pages/HDF5_structure.rst:91]: I missed that you already updated the documentation, sorry about that. Looks good.

**Comment by fschlueter** [comment]
> please don't switch to debug output. This is a useful output that let's the user check if the channels and noise temperatures and RMS values were calculated correctly.

Later the value is reported for one channel. Dont you think that is sufficient?

**Comment by cg-laser** [comment]
no, in many applications, different channels have different bandwidths

**Comment by fschlueter** [comment]
> no, in many applications, different channels have different bandwidths

What is the than of logging and storing the bandwidth of the first channel?

**Comment by fschlueter** [comment]
@cg-laser I improved the logging. Now, it should also be more obvious what the settings are in the config, i.e., whether the `Vrms` is calculated per channel based on a noise temperature and response or fixed to a certain value. 

For example this is the output for a couple of RNO-G channels:
```
STATUS:NuRadioMC:Station.channel 11.16: noise temperature = 300.0 K, est. bandwidth = 742.33 MHz, max. filter amplification = 8.10e+02 -> Vrms = 7.74 mV -> efield Vrms = 9.55muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.17: noise temperature = 300.0 K, est. bandwidth = 742.33 MHz, max. filter amplification = 8.10e+02 -> Vrms = 7.74 mV -> efield Vrms = 9.55muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.18: noise temperature = 300.0 K, est. bandwidth = 742.33 MHz, max. filter amplification = 8.10e+02 -> Vrms = 7.74 mV -> efield Vrms = 9.55muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.19: noise temperature = 300.0 K, est. bandwidth = 742.33 MHz, max. filter amplification = 8.10e+02 -> Vrms = 7.74 mV -> efield Vrms = 9.55muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.20: noise temperature = 300.0 K, est. bandwidth = 742.33 MHz, max. filter amplification = 8.10e+02 -> Vrms = 7.74 mV -> efield Vrms = 9.55muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.21: noise temperature = 300.0 K, est. bandwidth = 722.57 MHz, max. filter amplification = 1.77e+03 -> Vrms = 16.64 mV -> efield Vrms = 9.41muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.22: noise temperature = 300.0 K, est. bandwidth = 722.57 MHz, max. filter amplification = 1.77e+03 -> Vrms = 16.64 mV -> efield Vrms = 9.41muV/m (VEL = 1m)
STATUS:NuRadioMC:Station.channel 11.23: noise temperature = 300.0 K, est. bandwidth = 722.57 MHz, max. filter amplification = 1.77e+03 -> Vrms = 16.64 mV -> efield Vrms = 9.41muV/m (VEL = 1m)
STATUS:NuRadioMC:All signals with less then 2.0 x Vrms_efield will be skipped.
```

**Comment by fschlueter** [comment]
For my understanding the `trigger` settings in the config are not affecting the actual trigger simulations but only the speedup cuts? Is the name historically grown? Should we change that to avoid confusion?

**Comment by cg-laser** [comment]
Thanks Felix. I added the int. response to the output. Looks good to me now. 

About the `trigger` group in the config file. The noise temperature and Vrms influence the trigger threshold. So I think the naming still makes sense. And we shouldn't change it due to backward compatibility. You can always add comments to the default config file to clarify things.

**Comment by fschlueter** [comment]
@cg-laser a approval is needed here

**Review by cg-laser — APPROVED** [review]

**Pull request merged**